tests/src/FunctionalJavascript/Form/ElementTest.php

```diff
@@ -29,6 +30,15 @@ public function testOptgroups() {
 
     $this->assertSession()->elementTextContains('css', '.select2-results__group', 'Baba');
     $this->assertSession()->elementTextContains('css', 'ul.select2-results__options li.select2-results__option ul.select2-results__options--nested li.select2-results__option', 'Nana');
+
+    $page->find('css', '.select2-search__field')->setValue('Nana');
+    $page->find('css', '.select2-results__option--highlighted')->click();
+
+    $page->pressButton('Submit');
+
+    $json = json_decode($this->getSession()->getPage()->getText(), TRUE);
+
+    $this->assertArraySubset(['value' => 3], $json['select2_optgroups'][0]);
```

Comment:
It would be great to also have a test that:
- Modifies an exposed views filter to use the select2 element
- Ensures that when nothing is selected, the filter is not filtering out anything
- When an option is selected, it correctly restricts results according to what was selected

Reply:
Added test for all of that